feat: new annotation for exporting interfaces (#5565)
JetBrains/kotlin-web-site: docs/topics/js/js-to-kotlin-interop.md

It allows for direct mapping to regular TypeScript interfaces.

To export a Kotlin interface, for example from a Kotlin Multiplatform project:

1. Annotate the Kotlin interface with `@JsNoRuntime` in common code:

    ```kotlin
    // commonMain
    import kotlin.js.JsNoRuntime
    
    @JsNoRuntime
    expect interface DataProcessor {
        fun process(data: String): Int 
    }
    ```

2. Provide the actual implementation with `@JsNoRuntime` in your JS-specific source code:

    ```kotlin
    // jsMain
    import kotlin.js.JsNoRuntime
    
    @JsNoRuntime
    actual interface DataProcessor {
        actual fun process(data: String): Int
    } 
    ```
    
3. On the TypeScript side, the interface will be mapped to a regular TypeScript interface:
    
    ```typescript
    // Generated .d.ts
    export interface DataProcessor {
        process(data: string): number;
    }
    ```

For Kotlin Multiplatform projects, the general rules are:

* Both `expect` and `actual` interface declarations must be annotated with `@JsNoRuntime`. The only exception is
  `external` implementations in platform-specific code on the `actual` side.

> * Using `external` interface declarations in common code on the `expect` side is not recommended. Instead, use regular 

Reviewer: It's `forbidden` to use `expect external`

  interfaces annotated with `@JsNoRuntime`.

> Exporting Kotlin interfaces with `@JsNoRuntime` has some restrictions. The annotation is not allowed with:

Reviewer 2: ```suggestion
Exporting Kotlin interfaces with `@JsNoRuntime` has some restrictions. The annotation isn't allowed with:
```

> * `external` interfaces as they already behave as if they have `@JsNoRuntime` by default. Adding it will result in a compiler warning.

Reviewer 2: ```suggestion
* `external` interfaces as they already behave as if they have `@JsNoRuntime` by default. Adding it results in a compiler warning.
```
Future > Present tense.

* `is` and `as` type checks.
* Class references that use the [`::class` syntax](js-reflection.md).

> * Interfaces that are passed as reified type argument.

Reviewer 2: What about linking to this text? https://kotlinlang.org/docs/inline-functions.html#reified-type-parameters I know it's in inline functions, but I think it could be helpful if they really have no idea what it is.
